lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -5358,6 +5437,15 @@ impl<'a> FuncGen<'a> {
                                 ),
                             );
                         } else {
+                            let offset = this.assembler.get_offset().0;
+                            this.instructions_address_map.push(InstructionAddressMap {
+                                srcloc: SourceLoc::new(this.src_loc),
+                                code_offset: offset,
+                                code_len: 3,
```

Comment:
I got two bytes `ff d0`:

```
$ echo 'callq *%rax' | as -o x.o; objdump -d x.o

x.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <.text>:
   0:   ff d0                   callq  *%rax
```

There's a more complex `callq *(%rax, %rbx, 4)` which does encode to three bytes (`ff 14 98`) but I think singlepass doesn't emit it. It would be the base+disp case of `emit_call_location` if we do.

One alternative is to get the offset before and after `this.assembler.emit_call_location()` and use that to get the true length that was actually used.